gadgets/trace_malloc/program.bpf.c

```diff
@@ -0,0 +1,56 @@
+// SPDX-License-Identifier: GPL-2.0
+/* Copyright (c) 2024 The Inspektor Gadget authors */
+
+#include <vmlinux.h>
+#include <bpf/bpf_helpers.h>
+#include <bpf/bpf_tracing.h>
+#include <gadget/buffer.h>
+#include <gadget/macros.h>
+
+struct event {
+	__u32 pid;
+	__u64 addr;
+	bool is_malloc;
```

Comment:
I'm wondering if it's better to use an enum:. 

```c
enum type {
	MALLOC,
	FREE,
}
```

The name of the operation will be printed on the IU. And it'll also help if we want to add more functions like `calloc`, `realloc` in the future to this program.